src/components/vm/nics/nicAdd.tsx

```diff
@@ -201,25 +201,33 @@ export class AddNIC extends React.Component<AddNICProps, AddNICState> {
         }
 
         this.setState({ addVNicInProgress: true });
-        domainAttachIface({
-            connectionName: vm.connectionName,
-            vmName: vm.name,
-            model: this.state.networkModel,
-            sourceType: this.state.networkType,
-            source: this.state.networkSource,
-            sourceMode: this.state.networkSourceMode,
-            // Generate our own random MAC address because virt-xml has bug which generates different MAC for online and offline XML
-            // https://github.com/virt-manager/virt-manager/issues/305
-            mac: this.state.setNetworkMac ? this.state.networkMac : getRandomMac(vms),
-            permanent: this.state.permanent,
-            hotplug: vm.state === "running",
-        })
-                .then(() => {
-                    domainGet({ connectionName: vm.connectionName, id: vm.id });
-                    Dialogs.close();
-                })
-                .catch(exc => this.dialogErrorSet(_("Network interface settings could not be saved"), exc.message))
-                .finally(() => this.setState({ addVNicInProgress: false }));
+        try {
+            let source;
+            if (this.state.networkType == "direct") {
+                source = {
+                    "": this.state.networkSource,
+                    mode: this.state.networkSourceMode,
+                };
+            } else {
+                source = this.state.networkSource;
+            }
+            await virtXmlHotAdd(
+                vm,
+                "network",
+                {
+                    mac: this.state.setNetworkMac ? this.state.networkMac : getRandomMac(vms),
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-da07d010-20250929-141718-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)